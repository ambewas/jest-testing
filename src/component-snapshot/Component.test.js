// full docs https://hackernoon.com/testing-react-components-with-jest-and-enzyme-41d592c174f

import React from 'react';
import { shallow } from 'enzyme';
import MyComponent from './Component';

describe('Component', function () {
  const wrapper = shallow(
    <MyComponent>Hello Jest!</MyComponent>
  );

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should render a counter inside a div with classname child', () => {
    const child = wrapper.find('.child');

    expect(child).toBeDefined();
  })

  it('should increase a counter in state', () => {
    wrapper.find('.toggle').simulate('click');
    expect(wrapper.state().counter).toEqual(1)

    wrapper.find('.toggle').simulate('click');
    wrapper.find('.toggle').simulate('click');
    expect(wrapper.state().counter).toEqual(3)

  })
})
