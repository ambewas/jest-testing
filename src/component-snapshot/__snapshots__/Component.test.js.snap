// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should render correctly 1`] = `
<div
  className="toggle"
  onClick={[Function]}
  style={
    Object {
      "alignItems": "center",
      "cursor": "pointer",
      "display": "flex",
      "fontFamily": "sans-serif",
      "fontSize": 70,
      "height": "100vh",
      "justifyContent": "center",
      "userSelect": "none",
    }
  }
>
  <div
    className="child"
  >
    0
  </div>
  Hello Jest!
</div>
`;
